src/main/java/org/opensearch/geospatial/ip2geo/common/DatasourceFacade.java

```diff
@@ -137,6 +141,34 @@ public IndexResponse updateDatasource(final Datasource datasource) throws IOExce
         return client.index(requestBuilder.request()).actionGet(clusterSettings.get(Ip2GeoSettings.TIMEOUT));
     }
 
+    /**
+     * Delete datasource in an index {@code DatasourceExtension.JOB_INDEX_NAME}
+     *
+     * @param datasource the datasource
+     *
+     */
+    public void deleteDatasource(final Datasource datasource) {
+        if (client.admin()
+            .indices()
+            .prepareDelete(datasource.getIndices().toArray(new String[0]))
+            .setIndicesOptions(IndicesOptions.LENIENT_EXPAND_OPEN)
+            .execute()
+            .actionGet(clusterSettings.get(Ip2GeoSettings.TIMEOUT))
+            .isAcknowledged() == false) {
+            throw new OpenSearchException("failed to delete data[{}] in datasource", StringUtils.joinWith(",", datasource.getIndices()));
```

Comment:
nit: Why not use String.join method. My recommendation is to prefer jdk over external dependency